### PR TITLE
Fix assert's failed when clicking on a note

### PR DESCRIPTION
**Pull request by psmokotnin**
- Fix midi::Event assert's failed when clicking on a note

- fix playing a note on click

- fix CMake C++17 settings for macOS

### CODE REVIEW

**Comment by RobFog** [comment]
That should be the plural of "assert", so "asserts", right?

**Comment by psmokotnin** [comment]
@RobFog I meant here this assert: https://github.com/musescore/MuseScore/blob/72ca0c374e7de0531ff65ced4f8f3aa66809421d/framework/midi/midievent.h#L252

**Comment by RobFog** [comment]
Ah, just one assert then? I read the pull request title like there were multiple.

File: mu4/notation/internal/notationplayback.cpp [hunk 1]
```diff
@@ -453,12 +453,26 @@ MidiData NotationPlayback::playNoteMidiData(const Ms::Note* note) const
...
     Chunk chunk;
     chunk.beginTick = 0;
     chunk.endTick = Ms::MScore::defaultPlayDuration * 2;
-    chunk.events.insert({ chunk.beginTick, Event(channel, EventType::ME_NOTEON, pitch, 80) });
-    chunk.events.insert({ Ms::MScore::defaultPlayDuration, Event(channel, EventType::ME_NOTEOFF, pitch, 0) });
-    chunk.events.insert({ chunk.endTick, Event(channel, EventType::ME_EOT, 0, 0) });
+    auto event = Event(Event::Opcode::NoteOn);
+    event.setChannel(channel);
+    event.setNote(pitch);
+    event.setVelocityFraction(0.63); //as 80 for 127 scale
```
Comment (RomanPudashkin) [inline mu4/notation/internal/notationplayback.cpp:467]: This code is in 3 places. I think better to place it in one function and reuse it

**Pull request merged**